modules/programs/zsh/default.nix

```diff
@@ -107,15 +107,25 @@ in
       default = false;
       description = "Enable zsh-syntax-highlighting.";
     };
+
+    programs.zsh.enableFastSyntaxHighlighting = mkOption {
+      type = types.bool;
+      default = false;
+      description = lib.mdDoc "Enable zsh-fast-syntax-highlighting.";
+    };
   };
 
   config = mkIf cfg.enable {
 
+    warnings = mkIf (cfg.enableFastSyntaxHighlighting && cfg.enableSyntaxHighlighting) [
+      "zsh-fast-syntax-highlighting and zsh-syntax-highlighting are mutually exclusive. Disable one of them."
+    ];
```

Comment:
```suggestion
    assertions = [
      {
        assertion = !(cfg.enableSyntaxHighlighting && cfg.enableFastSyntaxHighlighting);
        message = "zsh-syntax-highlighting and zsh-fast-syntax-highlighting are mutually exclusive, please disable one of them.";
      }
    ];
```

Reply:
Thanks for the changes @Enzime I had forgotten about this commit.

Reply:
@Samasaur1 @Enzime has shown that Fast Syntax Highlighting is mutually exclusive with Zsh-syntax-highlighting and has provided alterations.